File: Data Structures & Algorithms/permutation-string/submission-0.cpp
class Solution {
public:
    bool checkInclusion(string s1, string s2) {
        vector<int>f1(26,0);
        vector<int>f2(26,0);
        for(auto v:s1) f1[v-'a']++;
        for(int right=0;right<s2.size();right++){
            f2[s2[right]-'a']++;
            if(right-(int)s1.size()>=0) f2[s2[right-s1.size()]-'a']--;
            if(f1==f2) return true;
        }
        return false;
    }

};

File: Data Structures & Algorithms/count-good-nodes-in-binary-tree/submission-0.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def goodNodes(self, root: TreeNode) -> int:
        ans =0
        def dfs(root,max_number=root.val):
            nonlocal ans
            if not root:
                return;
            if root.val>=max_number:
                ans=ans+1

            max_number=max(max_number,root.val)
            dfs(root.left,max_number)
            dfs(root.right,max_number)
        dfs(root)
        return ans

           